cypress/integration/v2/features/dynamic-content/site-data.spec.js

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * External dependencies
+ */
+import { registerTests } from '~stackable-e2e/helpers'
+
+describe( 'Dynamic Content - Site', registerTests( [
+	matchSiteData,
+	adjustFieldOptions,
+	adjustFieldValues,
+] ) )
+
+const fields = [ 'Site Title', 'Site Tagline', 'Site URL' ]
+const testString = [ 'Test Site Title', 'Test Site Tagline', 'Test Site URL' ]
+
+const adjustSiteField = ( fieldName, fieldOptions = {} ) => {
+	 cy.adjustDynamicContent( 'ugb/cta', 0, '.ugb-cta__title', {
+		 source: 'Site',
+		 fieldName,
+		 fieldOptions,
+	 } )
+}
+
+function matchSiteData() {
+	 it( 'should match dynamic content in site fields', () => {
+		 cy.setupWP()
+		 Object.keys( fields ).forEach( fieldName => {
+			 cy.newPost()
+			 cy.addBlock( 'ugb/cta' )
+			 cy.typePostTitle( `${ fields[ fieldName ] } Test` )
+
+			const fieldOptions = {}
+			adjustSiteField( fields[ fieldName ], fieldOptions )
+
+			//Asserting content value with assertBlockContent
+			cy.openInspector( 'ugb/cta', 'Style' )
+			if ( `${ fields[ fieldName ] }` === 'Site Title' ) {
+				cy
+					.selectBlock( 'ugb/cta' )
+					.assertBlockContent( '.ugb-cta__title', 'e2etest' )
+			} else if ( `${ fields[ fieldName ] }` === 'Site Tagline' ) {
+				cy
+					.selectBlock( 'ugb/cta' )
+					.assertBlockContent( '.ugb-cta__title', 'Just another WordPress site' )
+			} else if ( `${ fields[ fieldName ] }` === 'Site URL' ) {
+				cy
+					.selectBlock( 'ugb/cta' )
+					.assertBlockContent( '.ugb-cta__title', 'http://e2etest.local' )
+			}
```

Comment:
refactor this after restructuring `fields`. Let's not do `if` `else` / `else if` here